code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+	icon_state = "mining_rune"
+
+
+/obj/effect/warped_rune/adamantinespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/adamantinespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		auto_mining()
+
+
+/obj/effect/warped_rune/adamantinespace/proc/auto_mining()
+	for(var/turf/closed/mineral/ores in range(7,rune_turf)) //the range is pretty big to at least try to rival miners and their plasma cutters.
+		if(ores.mineralType != null) //here to counter runtimes when the mineral type of the rock is null
+			new ores.mineralType(rune_turf)
+
+
+/* Lightpink rune. Revive suicided/soulless corpses by yeeting a willing soul into it */
+
+
+/obj/item/slimecross/warping/lightpink
+	colour = "light pink"
+	runepath = /obj/effect/warped_rune/lightpinkspace
+	effect_desc = "draws a rune that will repair the soul of a humanoid corpse in the hope of bringing them back to life."
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/lightpinkspace
+	icon_state = "necro_rune"
+	desc = "Souls are like any other material, You just have to find the right place to manufacture them."
+	max_cooldown = 100
+
+
+/obj/effect/warped_rune/lightpinkspace/attack_hand(mob/living/user)
+	if(cooldown > world.time)
+		to_chat(user, "<span class='warning'>The rune is still charging!</span>")
+		return
+
+	for(var/mob/living/carbon/human/host in rune_turf)
+		if(!host.getorgan(/obj/item/organ/brain) || host.key || host.get_ghost(FALSE, TRUE)) //checks if the ghost and brain's there
+			to_chat(user, "<span class='warning'>This body can't be fixed by the rune in this state!</span>")
+			return
+
+		cooldown = world.time + max_cooldown //only start the cooldown if there's an actual body on there and it can be resurrected.
+		to_chat(user, "<span class='warning'>The rune is trying to repair [host.name]'s soul!</span>")
+		var/list/candidates = pollCandidatesForMob("Do you want to replace the soul of [host.name]?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, host,POLL_IGNORE_SENTIENCE_POTION)//sentience flags because lightpink.
+
+		if(length(candidates) && !host.mind) //check if anyone wanted to play as the dead person and check if no one's in control of the body one last time.
+			var/mob/dead/observer/ghost = pick(candidates)
+			host.key = ghost.key
+			host.suiciding = 0 //turns off the suicide var just in case
+			host.revive(full_heal = TRUE, admin_revive = TRUE) //might as well go all the way
+			to_chat(host, "<span class='warning'>You may wear the skin of someone else, but you know who and what you are. Pretend to be the original owner of this body as best as you can.</span>")
+			to_chat(user, "<span class='notice'>[host.name] is slowly getting back up with an empty look in [host.p_their()] eyes. It...worked?</span>")
+			playsound(host, "sound/magic/castsummon.ogg", 50, TRUE)
+			return
+
+		else
+			to_chat(user, "<span class='warning'>The rune failed! Maybe you should try again later.</span>")
```

Comment:
dust the corpse if you can't find any ghost candidates for it, so that leaving a corpse on a rune (or leaving MULTIPLE corpses on MULTIPLE runes) won't ping ghosts every minute until they select the "never" option for that prompt